lib/src/qr_code_scanner.dart

```diff
@@ -109,7 +108,7 @@ class _QRViewState extends State<QRView> {
         Container(
           padding: widget.overlayMargin,
           decoration: ShapeDecoration(
-            shape: widget.overlay,
+            shape: widget.overlay!,
```

Comment:
It's declared as ! because this function only gets called when widget.overlay != null.
We could change this to a `SizedBox.shrink()` however this will add an unnecessary `Stack`, `Container` and `ShapeDecoration` to the widget stack which i think is a bit messy.

```dart
return NotificationListener(
      onNotification: onNotification,
      child: SizeChangedLayoutNotifier(
        child: (widget.overlay != null)
            ? _getPlatformQrViewWithOverlay()
            : _getPlatformQrView(),
      ),
    );
```